File: searching/c/binarySearchUserInput.c
#include <stdio.h>

int binarySearch(int element[], int size,int key){
    int low = 0;
    int high = size - 1;

    while(low <= high){
        int middle = low +(high - low) / 2;

        while(low <= high){
            if (element[middle] == key)
            return middle;
        else if (element[middle] < key)
            low = middle + 1;
        else
            high = middle - 1;
        }
        return -1; // Return -1 if the element is not found
    };
};

int main(){
     int size, key;
    
    // Input the size of the array
    printf("Enter the size of the array: ");
    scanf("%d", &size);
    
    int arr[size];
    
    // Input array elements
    printf("Enter %d elements in sorted order:\n", size);
    for (int i = 0; i < size; i++) {
        printf("Element %d: ", i + 1);
        scanf("%d", &arr[i]);
    }
    
    // Input the key to search for
    printf("Enter the key to search for: ");
    scanf("%d", &key);
    
    int result = binarySearch(arr, size, key);
    
    if (result != -1)
        printf("Element %d is present at index %d\n", key, result);
    else
        printf("Element %d is not present in the array\n", key);
    
    return 0;
};

File: python/searching_algorithms/binary_search/binarySearchUserInput.py
def binarySearch(array,key,low,high):
    # Repeat until the pointers low and high meet each other
    while low <= high:
        middle = low + (high - low) // 2

        if array[middle] == key:
            return middle
        elif array[middle] < key:
            low = middle + 1
        else:
            high = middle - 1
    return -1


def main():
    size = int(input("Enter the size of the array: "))
    array = []

    print("Enter the element sequentially")
    # Input the elements of an Array
    for i in range(size):
        element = int(input("Element {}: ".format(i + 1)))
        array.append(element)

    key = int(input("Enter the key to search for: "))

    result = binarySearch(array,key,0,len(array) - 1)

    if(result == -1):
        print("The key {} is not found.".format(key))
    else:
        print("The key {} is found at index {}.".format(key, result))


if __name__ == "__main__":
    main()